packages/lib/src/components/FeedbackWidget/FeedbackWidget.tsx

```diff
@@ -0,0 +1,370 @@
+ */
+
+import {
+  feedbackFormSchema,
+  FeedbackFormType,
+  FeedbackInput,
+  OpenInput,
+  OpenInputTypeValues,
+  UserOutput,
+} from '@frachtwerk/essencium-types'
+import {
+  ActionIcon,
+  Box,
+  Button,
+  Center,
+  Dialog,
+  Flex,
+  Group,
+  Loader,
+  rem,
+  Stack,
+  Text,
+  Textarea,
+  Title,
+  Tooltip,
+  Transition,
+  UnstyledButton,
+  useMantineTheme,
+} from '@mantine/core'
+import { useDisclosure } from '@mantine/hooks'
+import {
+  IconBulb,
+  IconCameraCheck,
+  IconCameraPlus,
+  IconCircleCheck,
+  IconCircleX,
+  IconExclamationCircle,
+  IconMessageDots,
+} from '@tabler/icons-react'
+import html2canvas from 'html2canvas'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import { useEffect, useState } from 'react'
+import { Controller } from 'react-hook-form'
+
+import { useZodForm } from '../../hooks'
+
+type Props = {
+  createFeedback: (feedback: FeedbackInput) => void
+  currentUser: UserOutput | null
+  feedbackCreated: boolean
+  feedbackFailed: boolean
+  feedbackSending: boolean
+}
+
+export function FeedBackWidget({
+  createFeedback,
+  currentUser,
+  feedbackCreated,
+  feedbackFailed,
+  feedbackSending,
+}: Props): JSX.Element {
+  const theme = useMantineTheme()
+
+  const { t } = useTranslation()
+
+  const router = useRouter()
+
+  const [opened, { toggle, close }] = useDisclosure(false)
+
+  const [openInput, setOpenInput] = useState<OpenInputTypeValues | null>(null)
+
+  const [showSuccessMessage, setShowSuccessMessage] = useState<boolean>(false)
+
+  const [showErrorMessage, setShowErrorMessage] = useState<boolean>(false)
+
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+
+  const [screenshot, setScreenshot] = useState<string | null>(null)
+
+  const { handleSubmit, control, formState, reset } = useZodForm({
+    schema: feedbackFormSchema,
+  })
+
+  const iconStyling = {
+    size: openInput ? 16 : 40,
+    stroke: '1.5',
+  }
+
+  const icons = {
+    Issue: (
+      <IconExclamationCircle
+        size={iconStyling.size}
+        stroke={iconStyling.stroke}
+      />
+    ),
+    Idea: <IconBulb size={iconStyling.size} stroke={iconStyling.stroke} />,
+    Other: (
+      <IconMessageDots size={iconStyling.size} stroke={iconStyling.stroke} />
+    ),
+  }
+
+  useEffect(() => {
+    if (feedbackSending) {
+      setIsLoading(true)
+    }
+
+    if (feedbackCreated) {
+      setShowSuccessMessage(true)
+
+      setIsLoading(false)
+    }
+
+    if (feedbackFailed) {
+      setShowErrorMessage(true)
+
+      setIsLoading(false)
+    }
+  }, [feedbackCreated, feedbackFailed, feedbackSending])
+
+  function onCloseWidget(): void {
+    close()
+    reset()
+    setOpenInput(null)
+    setShowSuccessMessage(false)
+    setShowErrorMessage(false)
+    setIsLoading(false)
+    setScreenshot(null)
+  }
+
+  setTimeout(() => {
+    if (showSuccessMessage || showErrorMessage) {
+      onCloseWidget()
+    }
+  }, 3500)
```

Comment:
I think we have a better UX, if we just leave the success message after sending as is and the user must close the message manually. When opening again, the widget is reset. What do you think?

Reply:
I think both ways are fine, but I can remove the setTimeout function 🙂